airflow/providers/apache/spark/hooks/spark_sql.py

```diff
@@ -72,13 +77,33 @@ def __init__(
         executor_memory: Optional[str] = None,
         keytab: Optional[str] = None,
         principal: Optional[str] = None,
-        master: str = 'yarn',
+        master: Optional[str] = None,
         name: str = 'default-name',
         num_executors: Optional[int] = None,
         verbose: bool = True,
-        yarn_queue: str = 'default',
+        yarn_queue: Optional[str] = None,
     ) -> None:
         super().__init__()
+
+        try:
+            conn: "Optional[Connection]" = self.get_connection(conn_id)
+        except AirflowNotFoundException:
+            conn = None
+            options = {}
+        else:
+            options = conn.extra_dejson
+
+        # Set arguments to values set in Connection if not explicitly provided.
+        if master is None:
+            if conn is None:
+                master = "yarn"
+            elif conn.port:
+                master = f"{conn.host}:{conn.port}"
+            else:
+                master = conn.host
+        if yarn_queue is None:
+            yarn_queue = options.get("queue", "default")
```

Comment:
On the `self._yarn_queue = yarn_queue` bellow you can add the above logic like this:
```
self._yarn_queue = yarn_queue or options.get("queue", "default")
````

Reply:
Personally I prefer explcitly statements instead of `or`. I guess that’s subjective nad don’t have a strong opinion if there’s an established coding style. Is there?

Reply:
I also don't know if there is any coding style. My opinion is fewer lines and fewer variable changes.
I am happy if you want to keep it like that.